### PR TITLE
Function to filter trees from front end input

### PR DESCRIPTION
**Pull request by raustin9**
Co-authored-by: Trevor Mangrum <trevormangrum@users.noreply.github.com>

This creates a simple front end interface where they can choose how to filter trees.
A function then runs and filters the trees that will be displayed based on the input from the form submitted from the front end.

### CODE REVIEW

**Review by tbatk — APPROVED** [review]

**Review by rluberto — APPROVED** [review]
Looks great thanks for getting this done!

File: src/components/Filter/index.tsx [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+    const onChange = (event: React.SyntheticEvent) => {
+        event.persist();
+        const target = event.target as HTMLInputElement;
+
+        setValues(values => ({...values, [target.name]: target.value}));
+    }
+    const submit = async (event: React.SyntheticEvent) => {
+        event.preventDefault()
+    
+        const res = await fetch('/api/filter', {
```
Comment (andlrutt) [inline src/components/Filter/index.tsx:15]: Are we wanting to keep track of api routes in `utils/urls.ts?`

**Review by andlrutt — APPROVED** [review]
Looks great! @trevormangrum @rluberto Are we wanting to filter by a range of ages, or other fields, eventually? that might change how the query tree is passed.

File: src/components/Filter/index.tsx (hunk 1 above)
Comment (tbatk) [reply src/components/Filter/index.tsx:15]: Yes. Some things in next require using absolute routes, so having URL file makes it easy to implement those. It also makes the codebase easier to maintain in the event that one of the routes changes at any point.

**Comment by tbatk** [comment]
> Looks great! @trevormangrum @rluberto Are we wanting to filter by a range of ages, or other fields, eventually? that might change how the query tree is passed.

 I think this will be fine for now, but you're right, if we decide to add anything like that eventually,  we will have to change query tree.

**Pull request merged**